src/core/algorithms/nar/nar.cpp

```diff
@@ -0,0 +1,104 @@
+#include "nar.h"
+
+namespace model {
+
+std::string NAR::ToString() const {
+    auto map_to_string = [](auto const& map) {
+        std::ostringstream os;
+        bool first = true;
+        for (auto const& [key, range] : map) {
+            if (!first) {
+                os << ", ";
+            }
+            os << key << ": " << range->ToString();
+            first = false;
+        }
+        return os.str();
+    };
+
+    std::ostringstream result;
+    result << qualities_.fitness << " {";
+    result << map_to_string(ante_) << "} ===> {" << map_to_string(cons_);
+    result << "} s: " << qualities_.support << " c: " << qualities_.confidence;
+    return result.str();
+    return result.str();
```

Comment:
```suggestion
    return result.str();
```

Reply:
as far as I understand range based iota exists in std since C++ 23 and we use 20

https://en.cppreference.com/w/cpp/algorithm/ranges/iota

Reply:
Yes, sorry, I meant to say that we should use `feature_order.begin()` instead of `std::begin(feature_order)`, as that's how we're doing it everywhere else.